src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private T [] array;
+    private int arrayLength;
+    private int elementsAmount;
+
+    public ArrayList() {
+        arrayLength = DEFAULT_ARRAY_SIZE;
+        array = (T[]) new Object [arrayLength];
+    }
+
+    private void checkIndex(int index) {
+        if (index > elementsAmount || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound: " + index);
+        }
+        if (elementsAmount == arrayLength) {
+            resize(arrayLength);
+        }
```

Comment:
`resize `is unexpected to have into `checkIndex `method
remove resize from here

Reply:
This was mentor Ivan's offer "move this verification after checking the index".  Maybe I misunderstood him. 
Should i move it into "add" method.?